### DISCUSSION TITLE
Error on git push when host system is authenticated with `gh` cli

### DISCUSSION THREAD

**Discussion by MRDGH2821** [devcontainers/community#88] [Q&A]
This error occurs to me sometimes, not always. And usually rebuilding container does 
<details><summary>VS Code error on git push</summary>
<p>

![image](https://github.com/devcontainers/community/assets/46621001/96c114ca-22c7-4c91-b7e8-01a8373bc8e5)

</p>
</details> 

This is the git error log:
<details><summary>Git error log</summary>
<p>

```
> git push --follow-tags origin rewrite-sapphire:rewrite-sapphire
/usr/bin/gh auth git-credential get: 1: /usr/bin/gh: not found
node:internal/modules/cjs/loader:1078
  throw err;
  ^

Error: Cannot find module '/tmp/vscode-remote-containers-8f2fc77a-6da9-4b12-8ddd-7131bd3304b9.js'
    at Module._resolveFilename (node:internal/modules/cjs/loader:1075:15)
    at Module._load (node:internal/modules/cjs/loader:920:27)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)
    at node:internal/main/run_main_module:23:47 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}

Node.js v18.15.0
Missing or invalid credentials.
Error: connect ENOENT /tmp/user/1000/vscode-git-6819b2c906.sock
    at PipeConnectWrap.afterConnect [as oncomplete] (node:net:1494:16) {
  errno: -2,
  code: 'ENOENT',
  syscall: 'connect',
  address: '/tmp/user/1000/vscode-git-6819b2c906.sock'
}
Missing or invalid credentials.
Error: connect ENOENT /tmp/user/1000/vscode-git-6819b2c906.sock
    at PipeConnectWrap.afterConnect [as oncomplete] (node:net:1494:16) {
  errno: -2,
  code: 'ENOENT',
  syscall: 'connect',
  address: '/tmp/user/1000/vscode-git-6819b2c906.sock'
}
/usr/bin/gh auth git-credential erase: 1: /usr/bin/gh: not found
node:internal/modules/cjs/loader:1078
  throw err;
  ^

Error: Cannot find module '/tmp/vscode-remote-containers-8f2fc77a-6da9-4b12-8ddd-7131bd3304b9.js'
    at Module._resolveFilename (node:internal/modules/cjs/loader:1075:15)
    at Module._load (node:internal/modules/cjs/loader:920:27)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)
    at node:internal/main/run_main_module:23:47 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}

Node.js v18.15.0
remote: No anonymous write access.
fatal: Authentication failed for 'https://github.com/MRDGH2821/Discord-Ban-Utils-Bot.git/'
``` 

</p>
</details> 

As you can see in the logs, my repo is https://github.com/MRDGH2821/Discord-Ban-Utils-Bot/tree/rewrite-sapphire (`rewrite-sapphire` branch)

Here is my situation:
1. The repo is cloned in WSL
2. WSL is running on ubuntu
3. Docker is installed in host Windows with WSL integration enabled
4. Repo has `devcontainer.json`
5. VS Code uses this `devcontainer` config to open a container
6. Do some commits & push.

Most of the times it works flawlessly. But the above mentioned problem is very rare & today I finally grabbed screenshots of this.
Partly related to https://github.com/cli/cli/discussions/8344

**Comment by eljog** [comment]
Thank you so much for writing a detailed issue, we appreciate it.

From the details you have shared above and in https://github.com/cli/cli/discussions/8344 I see that `git` is trying to use `gh cli` for authentication, which obviously fails since the devcontainer do not have `gh cli` installed in it as we know.

So it is possibly an issue with the credential helper in vscode remote containers. Could you please open an issue at https://github.com/microsoft/vscode-remote-release/issues so that the vscode team can take a look at it?
